mido/midifiles/midifiles.py

```diff
@@ -373,18 +361,40 @@ def __iter__(self):
             raise TypeError("can't merge tracks in type 2 (asynchronous) file")
 
         tempo = DEFAULT_TEMPO
+        time_signature = DEFAULT_TIME_SIGNATURE
+        cum_delta = 0
         for msg in merge_tracks(self.tracks):
-            # Convert message time from absolute time
-            # in ticks to relative time in seconds.
+            # Convert relative message time to desired unit
             if msg.time > 0:
-                delta = tick2second(msg.time, self.ticks_per_beat, tempo)
+                if self.unit.lower() in ('t', 'ticks'):
+                    delta = msg.time
+                elif self.unit.lower() in ('s', 'sec', 'seconds'):
+                    delta = tick2second(msg.time, self.ticks_per_beat, tempo)
+                elif self.unit.lower() in ('b', 'beats'):
+                    delta = tick2beat(msg.time, self.ticks_per_beat,
+                                      time_signature)
+                else:
+                    raise ValueError("`unit` must be either 'ticks', 't', "
```

Comment:
The shorthands should be removed ('t', 's', 'b' are not obvious) as they provide very little benefit at a big cost. `MidiFile(unit='beats')` is a lot cleaner and easier to search in the docs after.

I.e. 

```
if self.unit.lower() == 'ticks':
```
is better.

Reply:
IMHO the supported units should be defined by an `Enum` with sensible names.
This would make the code clearer and simplify types verification.